### PR TITLE
drivers: ssd1673: add DT properties for config registers, add SSD1608 support

### PR DESCRIPTION
**Pull request by jfischer-no**
The values for the registers like GDV, SDV and Border Waveform
depend on the panel and display controller. Add DT properties
and obtain such values from DT.

Use compatible property to identify display and select
proper LUT for display controller.

Add support for SSD1608 controller and GDE029A1 display.

### CODE REVIEW

File: drivers/display/ssd1673.c [hunk 1]
```diff
@@ -467,20 +467,24 @@ static int ssd1673_controller_init(struct device *dev)
 		return err;
 	}
 
-	tmp[0] = SSD1673_VAL_GDV_CTRL_A;
-	tmp[1] = SSD1673_VAL_GDV_CTRL_B;
+	tmp[0] = DT_SOLOMON_SSD1673FB_0_CONTROLLER_GDV_A;
+#if defined(DT_SOLOMON_SSD1673FB_0_CONTROLLER_GDV_B)
+	tmp[1] = DT_SOLOMON_SSD1673FB_0_CONTROLLER_GDV_A;
```
Comment (vanwinkeljan) [inline drivers/display/ssd1673.c]: Should this not be GDV_B instead of GDV_A?

**Comment by vanwinkeljan** [comment]
@jfischer-phytec-iot definitely a good idea to put this info in DT.
Except the one small remark above I would like see the input from @galak  and @MaureenHelm on this topic as there is a proposal from @galak in PR https://github.com/zephyrproject-rtos/zephyr/pull/12457#issuecomment-457016502 on following the linux kernel way of handling display pannels.

File: drivers/display/ssd1673.c [hunk 2]
```diff
@@ -45,23 +45,35 @@ struct ssd1673_data {
...
+	0x00, 0x00, 0x00, 0x00, 0x00, 0x00
+};
+
+static u8_t ssd1673_lut_default[] = {
+	0x10, 0x18, 0x18, 0x08, 0x18, 0x18, 0x08, 0x00,
+	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+	0x00, 0x00, 0x00, 0x00, 0x13, 0x14, 0x44, 0x12,
+	0x00, 0x00, 0x00, 0x00, 0x00, 0x00
+};
+#endif
```
Comment (aurel32) [inline drivers/display/ssd1673.c]: Maybe add a #else and #error in case no LUT is defined for the current device.

**Review by aurel32 — COMMENTED** [review]
Thanks for adding support for multiple LUT. Besides the issue reported by @vanwinkeljan and a minor nitpick, it looks all good. I have added support for the Waveshare 1.54" display, you might want to fetch the commit https://github.com/aurel32/zephyr/commit/b682c87370bcc4aa0035f022922122c88115448d in your PR.

**Comment by aurel32** [comment]
> Except the one small remark above I would like see the input from @galak and @MaureenHelm on this topic as there is a proposal from @galak in PR [#12457 (comment)](https://github.com/zephyrproject-rtos/zephyr/pull/12457#issuecomment-457016502) on following the linux kernel way of handling display pannels.

I think the properties of a EPD and LCD or OLED display panels are quite different. Therefore I am not sure it applies here.

**Comment by jfischer-phytec-iot** [comment]
> > Except the one small remark above I would like see the input from @galak and @MaureenHelm on this topic as there is a proposal from @galak in PR [#12457 (comment)](https://github.com/zephyrproject-rtos/zephyr/pull/12457#issuecomment-457016502) on following the linux kernel way of handling display pannels.
> 
> I think the properties of a EPD and LCD or OLED display panels are quite different. Therefore I am not sure it applies here.

[proposal from @galak](https://github.com/zephyrproject-rtos/zephyr/pull/12457#issuecomment-457016502) could also fit EPDs. I think that what I did is not very clean :-)

**Comment by vanwinkeljan** [comment]
Looks like in PR #12457 panel specific parameters are moved into the config struct and set at build time based on selected panel

**Comment by jfischer-phytec-iot** [comment]
> Looks like in PR #12457 panel specific parameters are moved into the config struct and set at build time based on selected panel

How far is it better than to describe it in DT?

**Comment by vanwinkeljan** [comment]
> How far is it better than to describe it in DT?

Not much a difference the only advantage that I can see over DT is that there is no chance in making a typo in case you need the display on another board.

Any reason to keep the rfc label as the PR looks good to me?

**Comment by jfischer-phytec-iot** [comment]
>Any reason to keep the rfc label as the PR looks good to me?

removed RFC

**Review by aurel32 — APPROVED** [review]

**Review by vanwinkeljan — APPROVED** [review]

**Pull request merged**